server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/security/roles_controller.rb

```diff
@@ -0,0 +1,104 @@
+##########################################################################
+# Copyright 2016 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  module Admin
+    module Security
+      class RolesController < ::ApiV1::BaseController
+        before_filter :check_admin_user_and_401
+        before_action :check_for_stale_request, :check_for_attempted_rename, only: [:update]
+
+        def index
+          roles = service.listAll().to_a
+
+          if params[:role_type] == 'role'
+            roles.keep_if { |role| role.kind_of?(RoleConfig) }
+          elsif params[:role_type] == 'plugin_role'
```

Comment:
@jyotisingh @maheshp — should we explicitly call it `gocd_role` to distinguish from the common abstraction of a role?